src/openrct2/object/ObjectTypes.cpp

```diff
@@ -7,24 +7,73 @@
  * OpenRCT2 is licensed under the GNU General Public License version 3.
  *****************************************************************************/
 
+#include "ObjectTypes.h"
+
+#include "../util/Util.h"
 #include "Object.h"
 
 #include <algorithm>
 
+constexpr std::array kAllObjectTypes = {
+    ObjectType::Ride,
+    ObjectType::SmallScenery,
+    ObjectType::LargeScenery,
+    ObjectType::Walls,
+    ObjectType::Banners,
+    ObjectType::Paths,
+    ObjectType::PathAdditions,
+    ObjectType::SceneryGroup,
+    ObjectType::ParkEntrance,
+    ObjectType::Water,
+    ObjectType::ScenarioText,
+    ObjectType::TerrainSurface,
+    ObjectType::TerrainEdge,
+    ObjectType::Station,
+    ObjectType::Music,
+    ObjectType::FootpathSurface,
+    ObjectType::FootpathRailings,
+    ObjectType::Audio,
+};
+
+static_assert(kAllObjectTypes.size() == EnumValue(ObjectType::Count));
+
+// Object types that can be saved in a park file.
+static constexpr std::array<const ObjectType, kNumTransientObjectTypes> kTransientObjectTypes = {
+    ObjectType::Ride,         ObjectType::SmallScenery, ObjectType::LargeScenery,    ObjectType::Walls,
+    ObjectType::Banners,      ObjectType::Paths,        ObjectType::PathAdditions,   ObjectType::SceneryGroup,
+    ObjectType::ParkEntrance, ObjectType::Water,        ObjectType::TerrainSurface,  ObjectType::TerrainEdge,
+    ObjectType::Station,      ObjectType::Music,        ObjectType::FootpathSurface, ObjectType::FootpathRailings,
+};
+
+// Object types that cannot be saved in a park file.
+static constexpr std::array<const ObjectType, kNumIntransientObjectTypes> kIntransientObjectTypes = {
+    ObjectType::ScenarioText,
+    ObjectType::Audio,
+};
```

Comment:
I never quite understood why we need two tables for this, it is either excluded or included I'm fairly certain it can not be both at the same time, wouldn't be just one table sufficient and consider it excluded when not included?

Reply:
I'm more for a white list rather than black list tbh. It's a really small amount of elements so I doubt it would make any noticeable difference.

Reply:
I _think_ both lists are there in order to iterate over them. Though I guess that could be solved in other ways.

Reply:
I've given this a bit more thought. While I think it would be fine to change this to one list, reducing the potential for mistakes and thereby the maintenance burden, it would entail changing the logic quite a bit in the places that use either list. With this in mind, I think such a change is better suited to another PR.

Reply:
You are right, lets do this another time and not here.